src/freertos_drivers/net_cc32xx/CC32xxWiFi.hxx

```diff
@@ -305,6 +305,11 @@ public:
         return rssi;
     }
 
+    void add_ip_acquired_callback(std::function<void(bool)> callback)
```

Comment:
set_ip_acquired_callback
(normally an api called add_..._callback suggests there is a list of hooks, which is a common API, but you only have one)

please add documentation comment. Declare when this function will be called.
Declare which thread it will be called from!